pkg/cli/mirror/catalog_images.go

```diff
@@ -40,6 +41,56 @@ func (o *MirrorOptions) unpackCatalog(dstDir string, filesInArchive map[string]s
 	return found, nil
 }
 
+/*
+imageInfo contains file path and other metadata about an image
+*/
+type imageInfo struct {
+	imageDetails      []imageDetails // one or more details about an image (one per platform)
+	fullPathToRepoDir string         // path to the repo directory <some path>/src/catalogs/<repoPath>, where <repoPath> is a docker ref in filepath form
+}
+
+/*
+imageDetails captures catalog information for a given platform
+*/
+type imageDetails struct {
+	/*
+		a path to where an index.json file resides
+			single arch: <some path>/src/catalogs/<repoPath>/index/index.json
+			multi arch image: <some path>/src/catalogs/<repoPath>/multi/<platform>/index/index.json
+	*/
+	indexJsonPath string
+	platform      OperatorCatalogPlatform // platform associated with this image
+	hash          v1.Hash                 // the hash of this image pulled from the OCI layout directory
+}
+
+/*
+catalogsByImage keeps track of the file system paths for a given catalog.
+
+• key is catalog destination reference (with a tag and no digest),
+
+• value contains file system paths associated with this catalog as well as other metadata
+*/
+type catalogsByImage map[image.TypedImage]imageInfo
+
+/*
+rebuildCatalogs will modify an OCI catalog in <some path>/src/catalogs/<repoPath>/layout with
+the index.json files found in one or more of these locations:
+
+	single architecture image:    <some path>/src/catalogs/<repoPath>/index/index.json
+	multi architecture image(s):  <some path>/src/catalogs/<repoPath>/multi/<platform>/index/index.json
+
+# Arguments
+
+• ctx: cancellation context
+
+• dstDir: the path to where the config.SourceDir resides
+
+# Returns
+
+• image.TypedImageMapping: the source/destination mapping for the catalog (this contains all images for single or multiple architectures)
+
+• error: non-nil if error occurs, nil otherwise
+*/
 func (o *MirrorOptions) rebuildCatalogs(ctx context.Context, dstDir string) (image.TypedImageMapping, error) {
```

Comment:
Is it possible to add some unit tests , the cognitive complexity is high so we need to grasp this better